hooks/src/index.js

```diff
@@ -307,6 +311,16 @@ export function useErrorBoundary(cb) {
 	];
 }
 
+
+export function useId() {
+	const state = getHookState(currentIndex++, 11);
+	if (!state._id) {
+		currentIdCounter++;
+		state._id = '' + currentInternal._vnodeId + currentInternal._depth + currentIdCounter
```

Comment:
Personally I'm a bit hesitant on leveraging `_vnodeId` as it's less deterministic but I feel we need some root-identifier for `render`. `_depth` scares me a bit as well if components return different component trees (I.e. add some functional components) in server vs client. I think separating into `_domDepth` would resolve this, WDYT?